docs/en/stack/ml/nlp/ml-nlp-text-emb-vector-search-example.asciidoc

```diff
@@ -0,0 +1,339 @@
+    },
+    {
+      "set": {
+        "description": "Set error message",
+        "field": "ingest.failure",
+        "value": "{{_ingest.on_failure_message}}"
+      }
+    }
+  ]
+}
+--------------------------------------------------
+
+The passages are in a field named `text`. The `field_map` maps the text to the 
+field `text_field` that the model expects. The `on_failure` handler is set to 
+index failures into a different index.
+
+Before ingesting the data through the pipeline, create the mappings of the 
+destination index, in particular for the field `text_embedding.predicted_value` 
+where the ingest processor stores the embeddings. The msmarco-MiniLM-L-12-v3 model produces 
+embeddings with 384 dimensions, the `dense_vector` field must be configured 
+with the same number of dimensions as specified by the `dims` option.
+
+[source,js]
+--------------------------------------------------
+PUT collection-with-embeddings
+{
+  "mappings": {
+    "properties": {
+      "text_embedding.predicted_value": {
+        "type": "dense_vector",
+        "dims": 384,
+        "index": true,
+        "similarity": "cosine"
+      },
+      "text": {
+        "type": "text"
+      }
+    }
+  }
+}
+--------------------------------------------------
+
+Create the text embeddings by reindexing the data to the 
+`collection-with-embeddings` index through the {infer} pipeline. The {infer} 
+ingest processor inserts the embedding vector into each document.
+
+[source,js]
+--------------------------------------------------
+POST _reindex?wait_for_completion=false
+{
+  "source": {
+    "index": "collection"
+  },
+  "dest": {
+    "index": "collection-with-embeddings",
+    "pipeline": "text-embeddings"
+  }
+}
+--------------------------------------------------
+
+The API call returns a task ID that can be used to monitor the progress:
+
+[source,js]
+--------------------------------------------------
+GET _tasks/<task_id>
+--------------------------------------------------
+
+You can also open the model stat UI to follow the progress.
+
+[role="screenshot"]
+image::images/ml-nlp-text-emb-reindex.png[Model status UI]
+
+After the reindexing is finished, the documents in the new index contain the 
+{infer} results – the vector embeddings. 
+
+
+[discrete]
+[[ex-text-emb-vect-search]]
+== Vector similarity search
+
+To perform vector similarity search, you need to obtain the text embedding of a 
+text. This example uses the "How is the weather in Jamaica?" query as the input 
+text. The {ref}/infer-trained-model-deployment.html[_infer API] gives you the 
+embedding of this query as a dense vector:
+
+[source,js]
+--------------------------------------------------
+POST /_ml/trained_models/sentence-transformers__msmarco-minilm-l-12-v3/_infer
+{
+  "docs": {
+    "text_field": "How is the weather in Jamaica?"
+  }
+}
+--------------------------------------------------
+
+You can use the dense vector in the `collection-with-embeddings` index by doing 
```

Comment:
I think we're using the results from the previous API as the query_vector field in the following API. Can we make this a little clearer? For example:
```suggestion
You can use the resulting dense vector in the `query_vector` of a 
```